pkg/asset/rhcos/bootstrap_image.go

```diff
@@ -0,0 +1,58 @@
+// Package rhcos contains assets for RHCOS.
+package rhcos
+
+import (
+	"context"
+	"time"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/types/baremetal"
+)
+
+// BootstrapImage is location of the RHCOS image for the Bootstrap node
+// This stores the location of the image based on the platform.
+// eg. on AWS this contains ami-id, on Livirt this can be the URI for QEMU image etc.
+// Note that for most platforms this is the same as rhcos.Image
+type BootstrapImage string
+
+var _ asset.Asset = (*BootstrapImage)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BootstrapImage) Name() string {
+	return "BootstrapImage"
+}
+
+// Dependencies returns no dependencies.
+func (i *BootstrapImage) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate the RHCOS Bootstrap image location.
+func (i *BootstrapImage) Generate(p asset.Parents) error {
+	ic := &installconfig.InstallConfig{}
+	p.Get(ic)
+	config := ic.Config
+
+	var osimage string
+	var err error
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+	switch config.Platform.Name() {
+	case baremetal.Name:
+		// Baremetal IPI launches a local VM for the bootstrap node
+		// Hence requires the QEMU image to use the libvirt backend
```

Comment:
qq: the openstack image cannot be used to boot a VM using QEMU in libvirt backend? Is there a technical limitation?

Reply:
@abhinavdahiya  The reason is that the libvirt backend uses the qemu_fw_cfg driver to pass the Ignition config, and the only image which has this enabled is the QEMU one:

https://github.com/openshift/installer/blob/master/pkg/terraform/exec/plugins/vendor/github.com/dmacvicar/terraform-provider-libvirt/libvirt/domain.go#L243

It might be that in future we can rework that to enable passing the config via a config-drive so that we can use the openstack image in both places, but I've not investigated that yet, or had any discussion with the terraform-provider-libvirt community